cmd/logging/logging.go

```diff
@@ -0,0 +1,68 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// Config is passed to the binary logging function
+type Config struct {
+	ID        string
+	Namespace string
+	Stdout    io.Reader
+	Stderr    io.Reader
+}
+
+// LoggerFunc is a binary logging function signature
+type LoggerFunc func(context.Context, *Config, func() error) error
+
+// Run runs LoggerFunc
+func Run(fn LoggerFunc) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var errCh = make(chan error, 0)
+
+	sout, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_STDOUT"))
+	serr, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_STDERR"))
+	wait, _ := winio.DialPipeContext(ctx, os.Getenv("CONTAINER_WAIT"))
```

Comment:
We're ignoring the error values from these calls

Reply:
Might also be better to just assign the env variable lookups to variables and check if they even exist before trying to dial.

Reply:
I'm doing this intentionally, so that the binary will fail to start... so not sure how much defensive we want to be here?

Reply:
on the other hand it's probably better for the `LoggerFunc` to decide what to do with `nil` IOs. I'll also need to play around the `wait` channel and make sure that it's needed.